examples/taker_basic.rs

```diff
@@ -0,0 +1,191 @@
+//! Basic Taker API Example
+//!
+//! This example demonstrates how to use the Coinswap Taker API:
+//! - Initialize a Taker instance with Bitcoin Core RPC
+//! - Check wallet balance and generate addresses  
+//! - Sync with tracker and fetch maker offers
+//! - Find suitable makers and display offer details
+//! - Set up coinswap parameters
+//!
+//! ## Prerequisites
+//!
+//! Start Bitcoin Core in regtest mode:
+//! ```bash
+//! bitcoind -regtest -server -rpcuser=user -rpcpassword=password -rpcport=18443
+//! ```
+//!
+//! ## Usage
+//!
+//! ```bash
+//! cargo run --example taker_basic --features integration-test
+//! ```
+use bitcoin::Amount;
+use bitcoind::bitcoincore_rpc::Auth;
+use coinswap::{
+    taker::{SwapParams, Taker, TakerBehavior},
+    utill::ConnectionType,
+    wallet::RPCConfig,
+};
+use log::{error, info};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    println!("Coinswap Taker Basic Example");
+
+    // Setup data directory
+    let data_dir = std::env::temp_dir()
+        .join("coinswap_examples")
+        .join("taker_basic");
+    std::fs::create_dir_all(&data_dir)?;
+
+    let rpc_config = RPCConfig {
+        url: "127.0.0.1:18443".to_string(),
+        auth: Auth::UserPass("user".to_string(), "password".to_string()),
+        wallet_name: "".to_string(),
+    };
+
+    // Initialize Taker
+    let mut taker = Taker::init(
+        Some(data_dir),
+        None,
+        Some(rpc_config),
+        TakerBehavior::Normal,
+        None,
+        None,
+        Some(ConnectionType::CLEARNET),
+    )
+    .map_err(|e| format!("Failed to initialize Taker: {:?}", e))?;
+
+    println!("Taker initialized successfully");
+
+    // Check wallet balance and UTXOs
+    let wallet = taker.get_wallet();
+    let balances = wallet
+        .get_balances()
+        .map_err(|e| format!("Failed to get balances: {:?}", e))?;
+    let utxos = wallet
+        .get_all_utxo()
+        .map_err(|e| format!("Failed to get UTXOs: {:?}", e))?;
+
+    println!("Spendable: {} BTC", balances.spendable.to_btc());
+    println!("UTXOs: {}", utxos.len());
+
+    // Handle empty wallet
+    if balances.spendable == Amount::ZERO {
+        println!("Wallet is empty. Generate funding address:");
+        let wallet_mut = taker.get_wallet_mut();
+        let address = wallet_mut
+            .get_next_external_address()
+            .map_err(|e| format!("Failed to generate address: {:?}", e))?;
+        println!("Address: {}", address);
+        println!(
+            "Fund with: bitcoin-cli -regtest -rpcwallet=test sendtoaddress {} 0.01",
+            address
+        );
+        return Ok(());
+    }
+
+    // Wallet has funds - demonstrate offer fetching and maker discovery
+    println!("\nMaker Discovery:");
+    info!("Starting maker discovery");
```

Comment:
Recurring logs can be avoided. Just keep the println statements. Same for all logs below.